sql/expression_resolver.go

```diff
@@ -47,6 +47,15 @@ type engineSpec struct {
 	queue   string
 }
 
+type gitLabModule struct {
+	ModuleName   string
+	ProjectName  string
+	Sha          string
+	PrivateToken string
+	SourceRoot   string
+	GitLabServer string
```

Comment:
Maybe define some URI schema could reduce the number of parameters, like `token@https://gitlab.yourhost.com/your/repo.git:sha:/path/to/dir:ModelClassName`

Reply:
Is this a standard URL format for `gitlab`?

Reply:
Not a common standard, note we also use `maxcompute://xxx` to identify the database drivers. This is a method to define a pattern and reduce the number of paramters.